services/controllerhost/queueDepth.go

```diff
@@ -324,14 +325,20 @@ func (qdc *queueDepthCalculator) addExtentBacklog(
 
 func (qdc *queueDepthCalculator) computeBacklog(cgDesc *shared.ConsumerGroupDescription, cgExtent *metadata.ConsumerGroupExtent, storeMetadata *storeExtentMetadata, storeID string, logger bark.Logger) int64 {
 
-	var backlog int64
+	var backlog int64 = 0 // default
 	var iter = &qdc.iter
 
 	switch qdc.iter.isDLQ {
 	case true:
-		backlog = common.MaxInt64(0, storeMetadata.lastSequence-(storeMetadata.beginSequence+1))
+		// update backog, only if the begin/first seqnums are available
+		if storeMetadata.lastSequence != math.MaxInt64 && storeMetadata.beginSequence != math.MaxInt64 {
+			backlog = storeMetadata.lastSequence - storeMetadata.beginSequence + 1
+		}
 	case false:
-		backlog = common.MaxInt64(0, storeMetadata.availableSequence-cgExtent.GetAckLevelSeqNo())
+		// update backlog, only if there is an available seqnum
+		if storeMetadata.availableSequence != math.MaxInt64 {
+			backlog = storeMetadata.availableSequence - cgExtent.GetAckLevelSeqNo()
```

Comment:
I believe you still need max(0, diff) for backwards compatibility. If I understand right, the availableSequence can be -1 for empty extents with the old code.